Improved validation + user flow

djenriquez opened the pull request:
- Secrets must form a valid JSON before submitting
- Login config bug with submit fixed
- Login must have a Vault URL
- Canceling from login config will no longer save any settings

djenriquez committed:
Improved validation + user flow

- Secrets must form a valid JSON before submitting
- Login config bug with submit fixed
- Login must have a Vault URL
- Canceling from login config will no longer save any settings

Lucretius commented on app/components/Login/Login.jsx:
```diff
@@ -9,59 +9,55 @@ import MenuItem from 'material-ui/MenuItem';
+            errorMessage: "",
+            username: "",
+            password: "",
+            loginMethodType: window.localStorage.getItem("loginMethodType") || "GITHUB",
```
:+1:

Lucretius commented on app/components/Login/login.css:
```diff
@@ -17,5 +17,5 @@
 
 .error {
-    color:  rgb(244, 67, 54);
+    color:  red;
```
changing my red eh - that specific color matches the materialUI red color that comes with the framework (red500).  I think this is a different brand of red.

Lucretius commented on app/components/Login/Login.jsx:
```diff
@@ -72,137 +68,148 @@ export default class Login extends React.Component {
-            });
+                .then((resp) => {
+                    window.localStorage.setItem('vaultAccessTokenExpiration', Date.now() + (_.get(resp, 'data.lease_duration') * 1000))
+                    //DJ fill in the blanks here
```
any progress on actually supporting logins via username and password?

djenriquez replied:
Ah, I'll get that going in a separate PR.

djenriquez committed:
More validation

- Added error message when user tries to open a secret that does not
contain the Root Key specified in the settings
- Included full key name in delete confirmation message

djenriquez committed:
Update error red to recommended material-ui red
